subprojects/ide/src/integTest/groovy/org/gradle/plugins/ide/eclipse/EclipseProjectIntegrationTest.groovy

```diff
@@ -78,6 +78,540 @@ eclipse {
+                                <id>org.eclipse.some.custom.matcher</id>
+                                <arguments>barfoo</arguments>
+                        </matcher>
+                </filter>
+        </filteredResources>
+
+        '''
+        project.assertHasResourceFilterXml(resourceFilterXml)
+    }
+
+    void "existing project file with equivalent resource filters is unchanged"() {
+        given:
+        def projectFile = file('.project')
+        def projectFileOriginalText = '''<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>root</name>
+	<comment></comment>
+	<projects/>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+	</natures>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments/>
+		</buildCommand>
+	</buildSpec>
+	<linkedResources/>
+	<filteredResources>
+		<filter>
+			<id>1</id>
+			<type>9</type>
+			<name/>
+			<matcher>
+				<id>org.eclipse.some.custom.matcher</id>
+				<arguments>foobar</arguments>
+			</matcher>
+		</filter>
+		<filter>
+			<id>2</id>
+			<type>30</type>
+			<name/>
+			<matcher>
+				<id>org.eclipse.some.custom.matcher</id>
+				<arguments>barfoo</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
+'''
+
+        projectFile << projectFileOriginalText
+
+        and:
+        buildScript """
+apply plugin: 'java'
+apply plugin: 'eclipse'
+
+eclipse {
+  project {
+    // these filters are equivalent to those already in the .project
+    resourceFilter {
+      appliesTo = 'FOLDERS'
+      type = 'INCLUDE_ONLY'
+      recursive = false
+      matcher {
+        id = 'org.eclipse.some.custom.matcher'
+        arguments = 'foobar'
+      }
+    }
+    resourceFilter {
+      appliesTo = 'FILES_AND_FOLDERS'
+      type = 'EXCLUDE_ALL'
+      matcher {
+        id = 'org.eclipse.some.custom.matcher'
+        arguments = 'barfoo'
+      }
+    }
+  }
+}
+        """
+        when:
+        run("eclipse")
+        then:
+
+        def resourceFilterXml = '''
+        <filteredResources>
+                <filter>
+                        <id>1</id>
+                        <type>9</type>
+                        <name/>
+                        <matcher>
+                                <id>org.eclipse.some.custom.matcher</id>
+                                <arguments>foobar</arguments>
+                        </matcher>
+                </filter>
+                <filter>
+                        <id>2</id>
+                        <type>30</type>
+                        <name/>
+                        <matcher>
+                                <id>org.eclipse.some.custom.matcher</id>
+                                <arguments>barfoo</arguments>
+                        </matcher>
+                </filter>
+        </filteredResources>
+
+        '''
+        project.assertHasResourceFilterXml(resourceFilterXml)
+        projectFileOriginalText == projectFile.text
```

Comment:
This is the reason for the test failure. You need to normalize line separators if you want to compare a String and the content from a file.